db/document.go

```diff
@@ -1336,6 +1342,23 @@ func (doc *Document) UnmarshalWithXattrs(ctx context.Context, data, syncXattrDat
 			doc.SyncData = SyncData{}
 		}
 		doc._rawBody = data
+	case DocUnmarshalRevAndFlags:
+		// Unmarshal rev, cas and flags from sync metadata
+		if syncXattrData != nil {
+			var revOnlyMeta revAndFlagsSyncData
+			unmarshalErr := base.JSONUnmarshal(syncXattrData, &revOnlyMeta)
+			if unmarshalErr != nil {
+				return pkgerrors.WithStack(base.RedactErrorf("Failed to UnmarshalWithXattrs() doc with id: %s (DocUnmarshalRevAndFlags).  Error: %v", base.UD(doc.ID), unmarshalErr))
```

Comment:
nit: general go style

pkgerrors WithStack was used before go implemented its own error wrapping in go 1.13. Some of Sync Gateway's code base was written before this existed. Preferred way of doing this is to use `%w` on base.RedactErrorf or fmt.Errorf to use "native" go error wrapping.


```suggestion
				return base.RedactErrorf("Failed to UnmarshalWithXattrs() doc with id: %s (DocUnmarshalRevAndFlags).  Error: %w", base.UD(doc.ID), unmarshalErr)
```